### PR TITLE
Update model and add validation

### PR DESCRIPTION
**Pull request by HungSonSN**

### CODE REVIEW

File: app/models/learn.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Learn < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :word
+
+  validates :word_id, presence: true
```
Comment (dieunb) [inline app/models/learn.rb]: only use `validates :word, presence: true`

File: app/models/learn.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  validates :user_id, presence: true
```
Comment (dieunb) [inline app/models/learn.rb]: `validates :user, presence: true`

File: app/models/answer.rb [hunk 3]
```diff
@@ -1,3 +1,7 @@
 class Answer < ActiveRecord::Base
   belongs_to :word
+  has_many :results, dependent: :destroy
+
+  validates :v_word, presence: true
```
Comment (dieunb) [inline app/models/answer.rb]: `v_word`?

File: app/models/answer.rb [hunk 4]
```diff
@@ -1,3 +1,7 @@
...
+  validates :word_id, presence: true
```
Comment (dieunb) [inline app/models/answer.rb]: only use `validates :word, presence: true`

File: app/models/lesson.rb [hunk 5]
```diff
@@ -1,4 +1,9 @@
 class Lesson < ActiveRecord::Base
   belongs_to :user
   belongs_to :category
+  has_many   :results, dependent: :destroy
+
+  validates :name, presence: true
+  validates :user_id, presence: true
```
Comment (dieunb) [inline app/models/lesson.rb]: only `validates :user, presence: true`

File: app/models/lesson.rb [hunk 6]
```diff
@@ -1,4 +1,9 @@
...
+  validates :category_id, presence: true
```
Comment (dieunb) [inline app/models/lesson.rb]: only use `validates :category_id, presence: true`

File: app/models/result.rb [hunk 7]
```diff
@@ -2,4 +2,8 @@ class Result < ActiveRecord::Base
   belongs_to :lesson
   belongs_to :word
   belongs_to :answer
+
+  validates :lesson_id, presence: true
```
Comment (dieunb) [inline app/models/result.rb]: only use 
```
validates :lesson, presence: true
validates :word, presence: true
validates :answer, presence: true
```

File: app/models/user.rb [hunk 8]
```diff
@@ -1,2 +1,9 @@
 class User < ActiveRecord::Base
+  has_many :lessons, dependent: :destroy
+  has_many :learns, dependent: :destroy
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :name, presence: true, length: { maximum: 50 }
```
Comment (dieunb) [inline app/models/user.rb]: remove space before `}` and after `{`

File: app/models/user.rb [hunk 9]
```diff
@@ -1,2 +1,9 @@
...
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX },
```
Comment (dieunb) [inline app/models/user.rb]: remove space before `}` and after `{`

File: app/models/user.rb [hunk 10]
```diff
@@ -1,2 +1,9 @@
...
+                                uniqueness: { case_sensitive: false }
```
Comment (dieunb) [inline app/models/user.rb]: remove space before `}` and after `{`

File: app/models/word.rb [hunk 11]
```diff
@@ -1,3 +1,7 @@
 class Word < ActiveRecord::Base
   belongs_to :category
+  has_many   :learns, dependent: :destroy
+
+  validates :e_word, presence: true
+  validates :category_id, presence: true
```
Comment (dieunb) [inline app/models/word.rb]: only use 
```
validates :category, presence: true
```

File: app/models/word.rb [hunk 12]
```diff
@@ -1,3 +1,7 @@
 class Word < ActiveRecord::Base
   belongs_to :category
+  has_many   :learns, dependent: :destroy
+
+  validates :e_word, presence: true
```
Comment (dieunb) [inline app/models/word.rb]: `e_word`?

File: app/models/answer.rb (hunk 3 above)
Comment (nghiabk) [inline app/models/answer.rb]: v_word là từ tiếng anh có nghĩa tiếng việt anh ạ

File: app/models/word.rb (hunk 12 above)
Comment (nghiabk) [inline app/models/word.rb]: e_word là từ tiếng anh ạ

**Pull request merged**